storage/src/vespa/storage/storageserver/rpc/storage_api_rpc_service.cpp

```diff
@@ -117,7 +117,8 @@ void compress_and_add_payload_to_rpc_params(mbus::BlobRef payload,
 
     params.AddInt8(comp_type);
     params.AddInt32(static_cast<uint32_t>(to_compress.size()));
-    params.AddData(buf.stealBuffer(), buf.getDataLen());
+    auto buffer_len = buf.getDataLen();
+    params.AddData(buf.stealBuffer(), buffer_len);
```

Comment:
This is not the first time this treacherous interface gives us trouble.
Fixed in #14575